graphql-dgs-codegen-core/src/test/kotlin/com/netflix/graphql/dgs/codegen/CodeGenTest.kt

```diff
@@ -849,6 +848,170 @@ class CodeGenTest {
+            
+            type SearchResult {
+                person: Person
+            }
+            
+            type Person {
+                firstname: String
+                lastname: String
+                birthDate: Date
+            }
+        """.trimIndent()
+
+        val (dataTypes) = CodeGen(
+            CodeGenConfig(
+                schemas = setOf(schema),
+                packageName = basePackageName,
+                typeMapping = mapOf(Pair("Person", "mypackage.Person")),
+            )
+        ).generate()
+
+        assertThat(dataTypes.size).isEqualTo(1)
+        assertThat(dataTypes[0].typeSpec.fieldSpecs[0].type.toString()).isEqualTo("mypackage.Person")
+    }
+
+    @Test
+    fun `Use mapped type name when the type is mapped for interface`() {
+
+        val schema = """
+            type Query {                
+                search: SearchResult
+            }
+            
+            type SearchResult {
+                item: SomethingWithAName
+            }
+            
+            interface SomethingWithAName {
+                name: String
+            }
+            
+            type Person implements SomethingWithAName {
+                name: String
+            }
+        """.trimIndent()
+
+        val (dataTypes, javaInterfaces) = CodeGen(
+            CodeGenConfig(
+                schemas = setOf(schema),
+                packageName = basePackageName,
+                typeMapping = mapOf(
+                    Pair("SomethingWithAName", "mypackage.SomethingWithAName"),
+                    Pair("Person", "mypackage.Person"),
+                ),
+            )
+        ).generate()
+
+        assertThat(dataTypes.size).isEqualTo(1)
+        assertThat(dataTypes[0].typeSpec.fieldSpecs[0].type.toString()).isEqualTo("mypackage.SomethingWithAName")
+
+        assertThat(javaInterfaces).isEmpty()
+    }
+
+    @Test
+    fun `Use mapped type name when the type is mapped for union`() {
+
+        val schema = """
+            type Query {                
+                search: SearchResult
+            }
+            
+            union SearchResult = Actor | Movie
+                     
+            type Movie {
+                title: String
+            }
+            
+            type Actor {
+                name: String
+            }
+        """.trimIndent()
+
+        val (dataTypes, javaInterfaces) = CodeGen(
+            CodeGenConfig(
+                schemas = setOf(schema),
+                packageName = basePackageName,
+                typeMapping = mapOf(
+                    Pair("SearchResult", "mypackage.SearchResult"),
+                    Pair("Movie", "mypackage.Movie"),
+                    Pair("Actor", "mypackage.Actor"),
+                ),
+            )
+        ).generate()
+
+        assertThat(dataTypes).isEmpty()
+        assertThat(javaInterfaces).isEmpty()
+    }
+
+    @Test
+    fun `Use mapped type name when a concrete type of a union is mapped`() {
```

Comment:
similar to the type definition comments above, this would be easier to test if there was a shared component that each type definition encapsulated which exhibited this behavior